packages/cli/test/unit/commands/promote.test.ts

```diff
@@ -0,0 +1,380 @@
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput(
+      'Error: The provided argument "????" is not a valid deployment'
+    );
+    await expect(exitCodePromise).resolves.toEqual(1);
+  });
+
+  it('should error if deployment not found', async () => {
+    const { cwd } = initPromoteTest();
+    client.setArgv('promote', 'foo', '--yes', '--cwd', cwd);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput('Fetching deployment "foo" in ');
+    await expect(client.stderr).toOutput(
+      'Error: Error: Can\'t find the deployment "foo" under the context'
+    );
+
+    await expect(exitCodePromise).resolves.toEqual(1);
+  });
+
+  it('should show status when not rolling back', async () => {
+    const { cwd } = initPromoteTest();
+    client.setArgv('promote', '--yes', '--cwd', cwd);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput(
+      'Checking promotion status of vercel-promote'
+    );
+    await expect(client.stderr).toOutput('No deployment promotion in progress');
+
+    await expect(exitCodePromise).resolves.toEqual(0);
+  });
+
+  it('should promote by deployment id', async () => {
+    const { cwd, previousDeployment } = initPromoteTest();
+    client.setArgv('promote', previousDeployment.id, '--yes', '--cwd', cwd);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput(
+      `Fetching deployment "${previousDeployment.id}" in ${previousDeployment.creator?.username}`
+    );
+    await expect(client.stderr).toOutput('Promote in progress');
+    await expect(client.stderr).toOutput(
+      `Success! ${chalk.bold('vercel-promote')} was promoted to ${
+        previousDeployment.url
+      } (${previousDeployment.id})`
+    );
+
+    await expect(exitCodePromise).resolves.toEqual(0);
+  });
+
+  it('should promote by deployment url', async () => {
+    const { cwd, previousDeployment } = initPromoteTest();
+    client.setArgv('promote', previousDeployment.url, '--yes', '--cwd', cwd);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput(
+      `Fetching deployment "${previousDeployment.url}" in ${previousDeployment.creator?.username}`
+    );
+    await expect(client.stderr).toOutput('Promote in progress');
+    await expect(client.stderr).toOutput(
+      `Success! ${chalk.bold('vercel-promote')} was promoted to ${
+        previousDeployment.url
+      } (${previousDeployment.id})`
+    );
+
+    await expect(exitCodePromise).resolves.toEqual(0);
+  });
+
+  it('should get status while promoting', async () => {
+    const { cwd, previousDeployment, project } = initPromoteTest({
+      promotePollCount: 10,
+    });
+
+    // start the promote
+    client.setArgv('promote', previousDeployment.id, '--yes', '--cwd', cwd);
+    promote(client);
+
+    // need to wait for the promote request to be accepted
+    await sleep(500);
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏poll for the change

Instead of sleeping, can we poll (with a timeout) for the change we expect to see?

Reply:
We can't poll here in the test. It's testing the internal polling of the promote status. The promote command will poll every 250ms. Setting any lower things get flakey. I set it to 300ms.